projects/instant-ui-library/src/lib/loading-spinner/loading-spinner/loading-spinner.component.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  OnChanges,
+  SimpleChanges,
+} from '@angular/core';
+import { ModeType } from '../../shared/models/color.model';
+import { Size, SizeMode } from '../../shared/models/size.model';
+import { Mode } from './../../shared/models/color.model';
+
+@Component({
+  selector: 'iu-loading-spinner[id]',
+  templateUrl: './loading-spinner.component.html',
+  styleUrls: ['./loading-spinner.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LoadingSpinnerComponent implements OnChanges {
+  @Input() variant: ModeType;
+  @Input() sizeMode: SizeMode;
+  @Input() id: string = '';
+
+  spinnerSizeClass: string = '';
+  spinnerColorClass: string = '';
+  size = Size;
+  mode = Mode;
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes) {
+      switch (this.sizeMode) {
```

Comment:
Suggest using ngClass ability to conditionally set classes. This works because the property is immutable, but for scalability we could run into problems with change detections if we send mutable objects.

Reply:
Also this will cleanup the component

Reply:
Noice.